### ISSUE TITLE
Registering a host for the 2nd time fails when using force: false because of duplicate main interface

### ISSUE THREAD

**Issue by praffo** [ansible-collections/community.zabbix#244]
##### SUMMARY

Running the register tasks a second time fails because there cannot be more than one main interface, the interface has not changed

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
zabbix_host

##### ANSIBLE VERSION

```
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]

```

##### CONFIGURATION

```
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 30
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False

```

##### OS / ENVIRONMENT / Zabbix Version

Ubuntu 20.04.1 LTS on Windows WSL2

##### STEPS TO REPRODUCE

Register a new host
re run the same task

```
- name: Create a new host or update an existing host's info
  local_action:
      module: zabbix_host
      server_url: http://zabbix.example.com/
      login_user: "admin"
      login_password: "password"
      force: no
      host_name: "{{ inventory_hostname_short }}"
      host_groups:
        - Production
      link_templates:
        - "Custom Availability status"
        - "Template OS Linux by Zabbix agent active"
      interfaces:
        - type: 1 #Agent
          useip: 0 #Use DNS and not the ip
          main: 1 #Default interface
          dns: "{{ inventory_hostname_short }}"
          
      status: enabled
      state: present

```

##### EXPECTED RESULTS
Host information gets updated if there were any changes

##### ACTUAL RESULTS

Tasks fails with error: "msg": "Failed to update host host01: ('Error -32602: Invalid params., Host cannot have more than one default interface of the same type. while sending {"jsonrpc": "2.0", "method": "hostinterface.create", "params": {"main": 1, "type": 1, "useip": 0, "ip": "", "dns": "host01", "port": 10050, "details": [], "hostid": "10334"}, "auth": "0372669f0264079ce4e3e15f759067b9", "id": 16}', -32602)"

```
The full traceback is:
  File "/tmp/ansible_zabbix_host_payload_vv0vq_qn/ansible_zabbix_host_payload.zip/ansible/modules/monitoring/zabbix/zabbix_host.py", line 434, in update_host
  File "/usr/local/lib/python3.8/dist-packages/zabbix_api.py", line 341, in method
    return self.universal("%s.%s" % (self.data["prefix"], name), opts[0])
  File "/usr/local/lib/python3.8/dist-packages/zabbix_api.py", line 79, in wrapper
    return self.do_request(self.json_obj(method, opts))['result']
  File "/usr/local/lib/python3.8/dist-packages/zabbix_api.py", line 348, in do_request
    return self.parent.do_request(req)
  File "/usr/local/lib/python3.8/dist-packages/zabbix_api.py", line 299, in do_request
    raise ZabbixAPIException(msg, jobj['error']['code'])
fatal: [host01 -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ca_cert": null,
            "description": null,
            "force": false,
            "host_groups": [
                "Production"
            ],
            "host_name": "host01",
            "http_login_password": null,
            "http_login_user": null,
            "interfaces": [
                {
                    "dns": "host01",
                    "interfaceid": "12",
                    "ip": "",
                    "main": 1,
                    "port": "10050",
                    "type": 1,
                    "useip": 0
                },
                {
                    "details": [],
                    "dns": "host01",
                    "hostid": "10334",
                    "ip": "",
                    "main": 1,
                    "port": 10050,
                    "type": 1,
                    "useip": 0
                }
            ],
            "inventory_mode": null,
            "inventory_zabbix": null,
            "ipmi_authtype": null,
            "ipmi_password": null,
            "ipmi_privilege": null,
            "ipmi_username": null,
            "link_templates": [
                "Custom Availability status",
                "Template OS Linux by Zabbix agent active"
            ],
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_user": "praffo",
            "proxy": null,
            "server_url": "http://zabbix.example.com/",
            "state": "present",
            "status": "enabled",
            "timeout": 10,
            "tls_accept": 1,
            "tls_connect": 1,
            "tls_psk": null,
            "tls_psk_identity": null,
            "tls_subject": null,
            "validate_certs": true,
            "visible_name": null
        }
    },
    "msg": "Failed to update host host01: ('Error -32602: Invalid params., Host cannot have more than one default interface of the same type. while sending {\"jsonrpc\": \"2.0\", \"method\": \"hostinterface.create\", \"params\": {\"main\": 1, \"type\": 1, \"useip\": 0, \"ip\": \"\", \"dns\": \"host01\", \"port\": 10050, \"details\": [], \"hostid\": \"10334\"}, \"auth\": \"0372669f0264079ce4e3e15f759067b9\", \"id\": 16}', -32602)"
}

```

**Comment by D3DeFi** [comment]
@praffo thank you for opening this issue. I've tried to reproduce it with ansible 2.9.6 and Zabbix 5.0 and 5.2 without any luck.

Can you please provide what version of Zabbix are you running? This is somewhat a known issue (just a theoretical one as practically I've seen it pop up for the first time).

The problem is that we have interfaces that are assigned to an existing host and interfaces that user passed to the module. There is no way to match them except having them ordered for each type as any information in interfaces can be changed by the user. 

We cannot reliably use anything in interface for the comparison with any existing ones. E.g. lets say user wants to change both IP and DNS on 1st out of 2 agent type interfaces with port 10050 on the host. In such situation we simply cannot compare with existing interfaces and only need to rely on the fact that we modify 1st interface of type agent that Zabbix API returns.

This is simplified for situations where `force=yes` as we wipe any existing interfaces from the host when they don't match. However, there is an inconsistency between my theory and your bug report as it says you are just rerunning the same task 2 times and not changing anything in between module runs. Is that correct?

Can you consistently reproduce this issue? If so does is happen also with ansible 2.10?

**Comment by praffo** [comment]
Hi thanks for taking the time to look into this. I'm running Zabbix 5.0.3
and during this time I've switched computers and the new computer is
working as expected.

ansible --version
ansible 2.10.3
  config file = None
  configured module search path =
['/Users/praffo/.ansible/plugins/modules',
'/usr/share/ansible/plugins/modules']
  ansible python module location =
/usr/local/Cellar/ansible/2.10.3_1/libexec/lib/python3.9/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.9.0 (default, Dec  2 2020, 10:36:36) [Clang 12.0.0
(clang-1200.0.32.27)]

The difference I noted is that now there is a warning:
[WARNING]: The value "1" (type int) was converted to "'1'" (type string).
If this does not look like what you expect, quote the
entire value to ensure it does not change.

That warning goes away by changing interfaces: type: "1" (instead of 1),
this may be related to the issue

There is a small detail on the output:
changed: [host1] => {"changed": true, "result": "Successfully update host
host1 () and linked with template '['Custom Availability status', 'Template
OS Linux by Zabbix agent active']'"}

It said it changed the link_templates but the change was in the
host_groups, The changes were processed correclty

**Comment by D3DeFi** [comment]
I believe this can be closed as #309 was merged in and released with 1.2.0. 
It this problem arises again, please let us know
